Laboratory/Rust/functionality-testing/src/enum_size.rs

```diff
@@ -0,0 +1,10 @@
+use crate::generated::enum_size::SmallEnum;
+use crate::generated::enum_size::HugeEnum;
+use bebop::SubRecord;
+use std::mem::size_of;
+
+#[test]
+fn correct_sizes() {
+    assert_eq!(size_of::<SmallEnum>(), size_of::<u8>());
```

Comment:
This should technically be testing against `1` and `8` because if on a system u8 or i64 were different for any reason, it would cause the parsing to fail if it was encoded by a different system. IIRC these types have guaranteed sizes though so it is not really any different.

Reply:
Looks like that is the case:


> In general, the size of a type is not stable across compilations, but specific types such as primitives are.